source/Body.cpp

```diff
@@ -450,7 +451,7 @@ Body::doRHS()
 	// for rotation, this is just the negative of the body's rotation for now
 	// (not allowing flow rotation)
 	vec6 vi = -v6;
-	vi(Eigen::seqN(0, 3)) += U;
+	vi(Eigen::seqN(0, 3)) += U[0];
```

Comment:
Is this not wrong?

[U is already a vector](https://github.com/FloatingArrayDesign/MoorDyn/blob/v2/source/Body.hpp#L144). `vi(Eigen::seqN(0, 3)) += U;` means add `U` to the first 3 components of `vi`. AFAIK that was right

Reply:
As part of the waves refactor, structures now get the water kinematics from the waves class, and don't store it internally. In this case, a few lines above Body gets its wave kinematics with
```c++
auto [zeta, U, Ud] = waves->getWaveKinBody(bodyId);
// U is of type const std::vector<vec3>&, 
// but will have a size of 1 because a body only needs water kinematics at a single point
```
For simplicity reasons, all of the Wave::getWaveKin**** functions (for lines, connections, rods, bodies) return the water kinematics in the form of a vector for each "node". This includes for bodies and connections, even though they only have a single node. Doing it this way just reduces code duplication with minor variations inside of the Waves class. Because of this, bodies and connections do need to manually get the one element in the vector.

Reply:
Right, [you are dropping indeed U from the header](https://github.com/FloatingArrayDesign/MoorDyn/pull/71/files#diff-a76fe5fd1cca7a26ba246ab4d1461719534ccb31d1305dd71cd517f7f3ffc064L144)